README.md

```diff
@@ -1,2 +1,62 @@
 # install-scripts
 Installation scripts for OpenShift KNI clusters
+
+## CNV
+The CNV installation is managed by a 'meta operator' called the [HyperConverged
+Cluster Operator](https://github.com/kubevirt/hyperconverged-cluster-operator) (HCO).
+Deploying with the 'meta operator' will launch operators for KubeVirt,
+Containerized Data Imported (CDI), Cluster Network Addons (CNA),
+Common templates (SSP), KubeVirt Web UI, and Node Maintenance Operator (NMO).
+
+#### Deploy without OLM
+```bash
+$ curl https://raw.githubusercontent.com/kubevirt/hyperconverged-cluster-operator/master/deploy/deploy.sh | bash
+```
+
+#### Deploy CNV through Marketplace
+```bash
+$ curl https://raw.githubusercontent.com/kubevirt/hyperconverged-cluster-operator/master/tools/quay-registry.sh | CLUSTER=OPENSHIFT bash -s $QUAY_USERNAME $QUAY_PASSWORD
+```
+
+After pointing Marketplace at the new app registry, you can go the rest of the
+way using the UI by finding KubeVirt Hyperconverged Cluster Operator in the
+OperatorHub tab.
+
+If you want to use the CLI, run the following commands:
+
+```bash
+# Create a namespace for CNV
+kubectl create ns kubevirt-hyperconverged
+
+# Switch to the HCO namespace.
+kubectl config set-context $(kubectl config current-context) --namespace=kubevirt-hyperconverged
+
+# Create an OperatorGroup
+cat <<EOF | kubectl create -f -
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: hco-operatorgroup
+  namespace: kubevirt-hyperconverged
+EOF
+
+# Subscribe to the HCO
+cat <<EOF | kubectl create -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: hco-subscription
+  namespace: kubevirt-hyperconverged
+spec:
+  channel: alpha
+  name: kubevirt-hyperconverged
+  source: hco-catalogsource
+  sourceNamespace: openshift-operator-lifecycle-manager
+EOF
+
+# Wait for the HCO to appear
+oc wait pod $(oc get pods -n kubevirt-hyperconverged | grep hyperconverged-cluster-operator | head -1 | awk '{ print $1 }') --for condition=Ready -n kubevirt-hyperconverged --timeout="${POD_TIMEOUT}"
+
+# Create HCO's CR
+kubectl create -f https://raw.githubusercontent.com/kubevirt/hyperconverged-cluster-operator/master/deploy/converged/crds/hco.cr.yaml
```

Comment:
Is there an issue somewhere to have HCO create its own CR automatically?

Reply:
Do we actually want to couple this?
Tradtiionally (Ha!) operators are decoupled from the instnaces they create.

but maybe a nice idea ..

Reply:
There are a number of operators or controllers that are singletons. Like HCO, that type of operator only makes sense when there is exactly 1 of its CR. It's not the traditional operator pattern, but it is a variation that is reasonable to employ. For such an operator, using it involves two steps: Install the operator, then create the CR. For cases where the CR has sane defaults, or perhaps has no Spec at all, it's easier to just have the operator create its own CR on startup.

Reply:
I think an operator can go either way here.  Either the CR gets created automatically or have the admin create it as manual step.  My thinking on keeping it manual was that it maintains configuration as a day 1 operation as opposed to a day 2 operation.  In CNV, we currently don't have configuration that we're passing from the HCO down to the component operators, but we wanted to leave the option to be able to.  I image OCS will have some configuration and will probably want to keep the CR creation as a manual step also.